### PR TITLE
Convert ts from 'options' to 'task'

### PR DESCRIPTION
**Pull request by agubler**
This allows control on the temporary file (and clean up of the temporary files) and allows overriding of the initial properties (`grunt init`).

### CODE REVIEW

File: tasks/ts.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+		grunt.config.set('ts.options', baseOptions);
+
+		Object.keys(flags).forEach((key) => {
+			if (flags[key]) {
+				const tasks = [`ts:${key}`];
+				let tsconfigFileName = 'tsconfig.json';
+
+				// don't allow override of dev as uses projects tsconfig
+				if (key !== 'dev' && (defaultOptions[key] || tsOptions[key])) {
```
Comment (matt-gadd) [inline tasks/ts.ts]: I think we can `filter` this off after the `Object.keys`, which will neaten this code block up by removing one level of nesting

File: tasks/ts.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+					const tsconfigCopy = Object.assign(
+						{},
+						tsconfig,
+						defaultOptions[key],
+						tsOptions[key]
+					);
+					const compilerOptionsCopy = Object.assign(
+						{},
+						tsconfig.compilerOptions,
+						defaultOptions[key] ? defaultOptions[key].compilerOptions : {},
```
Comment (matt-gadd) [inline tasks/ts.ts]: hoisting these to a const will get rid of the repeated square bracket accessors

File: tasks/ts.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		const tsOptions = grunt.config.get<any>('ts');
+		const baseOptions = {
+			failOnTypeErrors: true,
+			fast: 'never'
+		};
+
+		grunt.config.set('ts.options', baseOptions);
+
+		Object.keys(flags).forEach((key) => {
+			if (flags[key]) {
```
Comment (matt-gadd) [inline tasks/ts.ts]: using `this.args` which returns an `array` means we won't need the Object.keys or this `if` condition which should neaten things up and remove a level of indentation.

**Comment by Tomdye** [comment]
👍

**Pull request merged**